src/backend/distributed/planner/insert_select_planner.c

```diff
@@ -909,7 +909,7 @@ RouterModifyTaskForShardInterval(Query *originalQuery,
 		 * Note that this is only the case with PG14 as the parameter doesn't exist
 		 * prior to that.
 		 */
-		shardRestrictionList = make_simple_restrictinfo(NULL,
+		shardRestrictionList = make_simple_restrictinfo(restriction->plannerInfo,
```

Comment:
can we merge this to main branch separately? It should be no-op for main. If so, please open a PR